addons/point_of_sale/wizard/pos_payment.py

```diff
@@ -7,59 +7,59 @@
 
 class PosMakePayment(models.TransientModel):
     _name = 'pos.make.payment'
-    _description = 'Point of Sale Payment'
+    _description = 'Point of Sale Make Payment Wizard'
 
-    def _default_session(self):
+    def _default_amount(self):
         active_id = self.env.context.get('active_id')
         if active_id:
-            return self.env['pos.order'].browse(active_id).session_id
+            return self.env['pos.order'].browse(active_id).amount_to_pay
         return False
 
-    def _default_journal(self):
+    def _default_payment_method(self):
         active_id = self.env.context.get('active_id')
         if active_id:
-            session = self.env['pos.order'].browse(active_id).session_id
-            return session.config_id.journal_ids and session.config_id.journal_ids.ids[0] or False
+            order_id = self.env['pos.order'].browse(active_id)
+            payment_method_ids = order_id.session_id.payment_method_ids
```

Comment:
You can do that instead:
return order_id.session_id.payment_method_ids.sorted(lambda pm: pm.is_cash_count, reverse=True)[:1]

sorted with reverse will put the elements having cash_count to True first, and [:1] will take the first element if any or False